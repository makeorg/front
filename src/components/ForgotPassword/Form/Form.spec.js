import jestExpect from 'expect';
import toMatchSnapshot from "expect-mocha-snapshot";
import renderer from 'react-test-renderer';
import ForgotPasswordFormComponent from './index';

describe("ForgotPasswordFormComponent", function () {
  it("must match the snapshot by default", function () {
    // TODO need to use shallow with jest for better testing
    const component = renderer.create(
      <ForgotPasswordFormComponent errors={[]} />
    );
    jestExpect(component).toMatchSnapshot(this);
  });

  it("must match the snapshot with errors", function () {
    // TODO need to use shallow with jest for better testing
    const component = renderer.create(
      <ForgotPasswordFormComponent errors={[{
        field: 'email',
        message: 'Error message 1'
      }, {
        field: 'field',
        message: 'Error message 2'
      }]} />
    );
    jestExpect(component).toMatchSnapshot(this);
  });
});