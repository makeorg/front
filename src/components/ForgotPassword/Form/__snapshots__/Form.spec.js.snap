// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordFormComponent must match the snapshot by default 1`] = `
.c4 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(232,24,68);
  background-color: rgb(232,24,68);
}

.c5 {
  margin-right: 0.375rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 1.125rem;
  border-radius: 1.875rem;
  border: 0.0625rem solid rgba(0,0,0,0.2);
  margin-bottom: 0.9375rem;
  background: rgb(242,242,242);
  background-color: rgb(242,242,242);
  border-color: rgba(0,0,0,0.2);
}

.c3 {
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 0.625rem;
  width: 100%;
  color: rgba(0,0,0,0.55);
  font-size: 0.875rem;
  line-height: 2.188rem;
}

.c2 {
  width: 1.875rem;
  color: rgb(232,24,68);
  font-size: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: 0.625rem 0 0;
  max-width: 30.63rem;
}

@media (min-width:47.94rem) {
  .c4 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c4 {
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c5 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c3 {
    font-size: 1rem;
    line-height: 2.5rem;
  }
}

@media (min-width:47.94rem) {
  .c2 {
    font-size: 1.125rem;
  }
}

<form
  className="c0"
  id="forgot_password"
>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="email"
    >
      <svg
        aria-hidden={true}
        className="svg-inline--fa fa-envelope fa-w-16 "
        data-icon="envelope"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </label>
    <input
      aria-required={true}
      className="c3"
      id="email"
      name="email"
      required={true}
      tabIndex={-1}
      type="email"
    />
  </div>
  <button
    className="c4"
    form="forgot_password"
    tabIndex={-1}
    type="submit"
  >
    <span
      className="c5"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-paper-plane fa-w-16 "
        data-icon="paper-plane"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M440 6.5L24 246.4c-34.4 19.9-31.1 70.8 5.7 85.9L144 379.6V464c0 46.4 59.2 65.5 86.6 28.6l43.8-59.1 111.9 46.2c5.9 2.4 12.1 3.6 18.3 3.6 8.2 0 16.3-2.1 23.6-6.2 12.8-7.2 21.6-20 23.9-34.5l59.4-387.2c6.1-40.1-36.9-68.8-71.5-48.9zM192 464v-64.6l36.6 15.1L192 464zm212.6-28.7l-153.8-63.5L391 169.5c10.7-15.5-9.5-33.5-23.7-21.2L155.8 332.6 48 288 464 48l-59.4 387.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </button>
</form>
`;

exports[`ForgotPasswordFormComponent must match the snapshot with errors 1`] = `
.c6 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  font-family: 'Roboto Condensed Bold',Arial,serif;
  font-weight: bold;
  font-size: 0.875rem;
  border: none;
  border-radius: 1.875rem;
  text-transform: uppercase;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.5);
  padding: 0.3125rem 0.9375rem;
  color: rgb(255,255,255);
  background: rgb(232,24,68);
  background-color: rgb(232,24,68);
}

.c7 {
  margin-right: 0.375rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 0 1.125rem;
  border-radius: 1.875rem;
  border: 0.0625rem solid rgba(0,0,0,0.2);
  margin-bottom: 0.9375rem;
  background: rgb(242,242,242);
  background-color: rgb(242,242,242);
  border-color: rgba(0,0,0,0.2);
}

.c5 {
  border: none;
  background: transparent;
  background-color: transparent;
  padding: 0 0.625rem;
  width: 100%;
  color: rgba(0,0,0,0.55);
  font-size: 0.875rem;
  line-height: 2.188rem;
}

.c4 {
  width: 1.875rem;
  color: rgb(232,24,68);
  font-size: 1rem;
}

.c1 {
  font-size: 0.75rem;
  margin-top: 0.625rem;
  margin-bottom: 0.625rem;
  color: rgb(237,24,68);
}

.c2 {
  list-style: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  margin: 0.625rem 0 0;
  max-width: 30.63rem;
}

@media (min-width:47.94rem) {
  .c6 {
    font-size: 1rem;
  }
}

@media (min-width:47.94rem) {
  .c6 {
    padding: 0.5625rem 1.25rem;
  }
}

@media (min-width:47.94rem) {
  .c7 {
    margin-right: 0.625rem;
  }
}

@media (min-width:47.94rem) {
  .c5 {
    font-size: 1rem;
    line-height: 2.5rem;
  }
}

@media (min-width:47.94rem) {
  .c4 {
    font-size: 1.125rem;
  }
}

<form
  className="c0"
  id="forgot_password"
>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      Error message 1
    </li>
    <li
      className="c2"
    >
      Error message 2
    </li>
  </ul>
  <div
    className="c3"
  >
    <label
      className="c4"
      htmlFor="email"
    >
      <svg
        aria-hidden={true}
        className="svg-inline--fa fa-envelope fa-w-16 "
        data-icon="envelope"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </label>
    <input
      aria-required={true}
      className="c5"
      id="email"
      name="email"
      required={true}
      tabIndex={-1}
      type="email"
    />
  </div>
  <button
    className="c6"
    form="forgot_password"
    tabIndex={-1}
    type="submit"
  >
    <span
      className="c7"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-paper-plane fa-w-16 "
        data-icon="paper-plane"
        data-prefix="far"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M440 6.5L24 246.4c-34.4 19.9-31.1 70.8 5.7 85.9L144 379.6V464c0 46.4 59.2 65.5 86.6 28.6l43.8-59.1 111.9 46.2c5.9 2.4 12.1 3.6 18.3 3.6 8.2 0 16.3-2.1 23.6-6.2 12.8-7.2 21.6-20 23.9-34.5l59.4-387.2c6.1-40.1-36.9-68.8-71.5-48.9zM192 464v-64.6l36.6 15.1L192 464zm212.6-28.7l-153.8-63.5L391 169.5c10.7-15.5-9.5-33.5-23.7-21.2L155.8 332.6 48 288 464 48l-59.4 387.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </button>
</form>
`;
